### DISCUSSION TITLE
Using the filesystem for additional caching?

### DISCUSSION THREAD

**Discussion by mathgeekcoder** [ambros-gleixner/MIPcc23#14] [Q&A] [answered]
Hi!

I'm curious if we are allowed to store additional solve information (i.e., a cache) to the file system?  And if so, what are the file size limits?

I realize we can add extra lines to the output (with # comment), but I am thinking a separate file that is structured for fast look ups.  I'm mainly asking in case we hit issues with the 16GB memory limit, but also in practice there might be long periods of time between instances where you don't want to keep everything in memory.

Thanks!
Luke

**Comment by matbesancon** [comment, marked as answer]
Hi Luke,

> I'm curious if we are allowed to store additional solve information (i.e., a cache) to the file system

yes absolutely, it will indeed be more convenient than storing things in the comments of the output. There shouldn't be any problems with file size limits up to "reasonable size", on the order of a (few) GB. With this you can also use mmap approaches to look up information. I'll get back to you if you need specific numbers but I'd be surprised if this is a limiting factor

**Reply by mathgeekcoder** [reply]
Wonderful!  Thanks for your reply.  

I'll need to experiment to see exactly how much storage we'll realistically need, so I might follow back with a request for specific numbers if I think we are getting 'unreasonably' large.

Thanks again!
Luke
